src/foam/graphics/TreeNode.js

```diff
@@ -0,0 +1,297 @@
+/**
+ * @license
+ * Copyright 2019 The FOAM Authors. All Rights Reserved.
+ * http://www.apache.org/licenses/LICENSE-2.0
+ */
+
+foam.CLASS({
+  package: 'foam.graphics',
+  name: 'TreeNode',
+  extends: 'foam.graphics.Box',
+
+  requires: [
+    'foam.graphics.Box',
+    'foam.graphics.CView',
+    'foam.graphics.Label',
+    'foam.graphics.Line'
+  ],
+
+  imports: [
+    'formatNode',
+    'graph',
+    'nodeHeight',
+    'nodeWidth',
+    'padding',
+    'parentNode?',
+    'relationship'
+  ],
+  exports: ['as parentNode'],
+
+  properties: [
+    'data',
+    {
+      name: 'outline',
```

Comment:
This variable could use some documentation describing that it's an array containing the max left and right values per level of the tree.

Reply:
Added.